lib/Optimizer/GraphOptimizer/PassManager.cpp

```diff
@@ -137,6 +137,11 @@ bool FunctionPassManager::runPrePass(Function *F,
   if (verifyBeforeAllPassesOpt ||
       listContainsString(verifyBeforePassesOpt, P.getName())) {
     CHECK(F->verify());
+    if (F->getState() >= FunctionState::FuncLoaded && backend_) {
```

Comment:
Is there a reason we need to check the FunctionState here? I think it's useful for stopping destructive optimizations like Constant deletion from the Module, but for verification it's fine to do before the Function is loaded.

Reply:
That's fair, removed it for now, I was thinking of my next step: having a solver add layout constraints instead of doing them in the loader, in that case we might fail upcoming verifications.